scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -0,0 +1,399 @@
+      return fileTime.toInstant();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Grab the completion time from a log file
+   *
+   * @param file we are checking
+   * @return completion time
+   */
+  private static Instant getLogCompletionTime(File file) {
+    try {
+      Scanner myReader = new Scanner(file);
+      Pattern pattern =
+          Pattern.compile("^\\[INFO] (.*) \\[.*] exiting with return code \\d+" + " .*");
+      while (myReader.hasNextLine()) {
+        Matcher matcher = pattern.matcher(myReader.nextLine());
+        if (matcher.matches()) {
+          String stringDate = matcher.group(1);
+          String datePattern = "yyyy-MM-dd HH:mm:ss[.SSS][,SSS][Z][X]";
+          DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(datePattern);
+          return LocalDateTime.parse(stringDate, dateTimeFormatter)
+              .atOffset(ZoneOffset.UTC)
+              .toInstant();
+        }
+      }
+    } catch (FileNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+    return null;
+  }
+
+  /**
+   * @param runDirectory which we are currently processing
+   * @param smrtCellCount total number of SMRT cells in runDirectory
+   * @return true or false
+   */
+  private boolean isRunComplete(File runDirectory, int smrtCellCount) {
+    try (Stream<Path> testFileStream = Files.walk(runDirectory.toPath());
+        Stream<Path> testDoneStream = Files.walk(runDirectory.toPath())) {
+      List<Path> transferTestFiles =
+          testFileStream
+              .filter(Files::isRegularFile)
+              .filter(file -> TRANSFER_TEST.test(String.valueOf(file.getFileName())))
+              .toList();
+
+      List<Path> transferDoneFiles =
+          testDoneStream
+              .filter(Files::isRegularFile)
+              .filter(file -> TRANSFER_DONE.test(String.valueOf(file.getFileName())))
+              .toList();
+
+      return transferDoneFiles.size() == smrtCellCount && transferTestFiles.size() == smrtCellCount;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Check for presence of Transfer_Test file and grab creation time
+   *
+   * @param runDirectory which we are currently processing
+   * @param dto PacBioNotificationDto which we will modify
+   */
+  private void setStartTimeFromTransferTest(File runDirectory, PacBioNotificationDto dto) {
+    Optional<Instant> earliestTransferTestCreationTime =
+        streamSmrtCellSubdirectories(runDirectory, "metadata")
+            .flatMap(
+                metadataDirectory ->
+                    Stream.of(
+                        metadataDirectory.listFiles(file -> TRANSFER_TEST.test(file.getName()))))
+            .map(RevioPacBioProcessor::getFileCreationTime)
+            .min(Comparator.naturalOrder());
+    // Set run start time based on most recently created .Transfer_Test file
+    dto.setStartDate(earliestTransferTestCreationTime.get());
```

Comment:
Will throw `NoSuchElementException` if the files are missing. That's probably fine for now, but you should document it here